src/Emulator/Host/fs.cpp

```diff
@@ -0,0 +1,33 @@
+#include "fs.h"
+#include <algorithm>
+
+namespace Emulator::Host::Fs {
+void MntPoints::mount(const std::string& host_folder, const std::string& guest_folder) { 
+	Lib::LockMutexGuard lock(m_mutex);
+
+	MntPair pair;
+    pair.host_path = host_folder;
+    pair.guest_path = guest_folder;
+
+	m_mnt_pairs.push_back(pair);
+}
+void MntPoints::unMount(const std::string& path) {}
+void MntPoints::unMountAll() {}
+std::string MntPoints::getHostDirectory(const std::string& guest_directory) { 
+	Lib::LockMutexGuard lock(m_mutex);
+    for (auto& pair : m_mnt_pairs)
+    {
+        if (pair.guest_path.starts_with(guest_directory)) {
+            return pair.host_path + guest_directory;
```

Comment:
There is a hidden vulnerability here. If a PS4 path uses `..` like this: `/app0/../../..` it can escape the emulator filesystem completely and touch any system file

Reply:
Relevant Panda code: https://github.com/wheremyfoodat/Panda3DS/blob/7c33ff59cd91927fa08a308cad1b87f526e0f069/include/fs/archive_base.hpp#L174-L218

Reply:
oh god can do that my own can you help me?:D

Reply:
ΟΚ